packages/react-juce/src/components/Button.tsx

```diff
@@ -84,14 +86,26 @@ export class Button extends Component<ButtonProps, ButtonState> {
     }
   }
 
+  handleEnter = (e: SyntheticMouseEvent): void => {
```

Comment:
For cases like this one where we only wrap the supplied `props.onMouseEnter` in a function which checks that the thing is callable, we can just omit this code.

MouseDown, MouseUp, and onClick are special cases where we have to do some juggling for the end user before we call their provided callbacks. MouseEnter and MouseLeave are straightforward enough that we can just let them through to the underlying `View` if they're present.

So that means that where we write:
```
const { onMouseDown, onMouseUp, onClick, ...other } = this.props;
```
onMouseLeave and OnMouseEnter will get wrapped up into `...other` if they're present, and then get unwrapped via `{...other}` onto the View on line 111. If these props are not available, they won't get wrapped/unwrapped and passed on, which is just as we want.

So basically you can omit all the changes to this file except for 2 line change to the typescript interface above

Reply:
Ahh yeah good shout. Skimmed too quickly.

Reply:
Oh okay I see, thanks for the review :) I noticed the more complex behaviour with the other events but didn't realize it wasn't needed to explicitly declare these ones, but the explanation is clear! 

I'll clean this and commit the changes!

Reply:
It's done! I'll let you check by principle but I tested and it still works without the useless code :)